File: DSA/Hima-Swarupa-Padala/day17q2.py
class TreeNode:    
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def insert_into_BST(root, value):
    if not root:
        return TreeNode(value)  

    if value > root.val:
        root.right = insert_into_BST(root.right, value)  
    elif value < root.val:
        root.left = insert_into_BST(root.left, value) 

    return root

root = TreeNode(4)
root.left = TreeNode(2)
root.right = TreeNode(7)
root.left.left = TreeNode(1)
root.left.right = TreeNode(3)

new_value = 5
root = insert_into_BST(root, new_value)

def inorder_traversal(node):
    if not node:
        return
    inorder_traversal(node.left)
    print(node.val, end=" ")
    inorder_traversal(node.right)

inorder_traversal(root)

File: DSA/Aarohi-Sharma/Day17_q2.cpp
#include <iostream>
using namespace std;
class TreeNode {
public:
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        if (root == NULL) {
            return new TreeNode(val); 
        }
        if (val > root->val) {
            root->right = insertIntoBST(root->right, val);
        } 
        else {
            root->left = insertIntoBST(root->left, val);
        }
        return root;
    }
};
void inorderTraversal(TreeNode* root) {
    if (root != NULL) {
        inorderTraversal(root->left); 
        cout << root->val << " "; 
        inorderTraversal(root->right); 
    }
}
int main() {
    Solution solution;
    TreeNode* root = new TreeNode(4);
    root->left = new TreeNode(2);
    root->right = new TreeNode(7);
    root->left->left = new TreeNode(1);
    root->left->right = new TreeNode(3);
    cout << "Inorder traversal of the original BST: ";
    inorderTraversal(root);
    cout << endl;
    int val = 5;
    root = solution.insertIntoBST(root, val);
    cout << "Inorder traversal of the BST after inserting " << val << ": ";
    inorderTraversal(root);
    cout << endl;
    return 0;
}
